backend/src/server/routes/v3/dashboard-router.ts

```diff
@@ -173,108 +174,114 @@ export const registerDashboardRouter = async (server: FastifyZodProvider) => {
-                event: {
-                  type: EventType.GET_SECRETS,
-                  metadata: {
-                    environment,
-                    secretPath,
-                    numberOfSecrets: secretCountFromEnv
-                  }
-                }
-              });
+          if (remainingLimit > 0 && totalSecretCount > adjustedOffset) {
+            secrets = await server.services.secret.getSecretsRawMultiEnv({
+              actorId: req.permission.id,
+              actor: req.permission.type,
+              actorOrgId: req.permission.orgId,
+              environments,
+              actorAuthMethod: req.permission.authMethod,
+              projectId,
+              path: secretPath,
+              orderBy,
+              orderDirection,
+              search,
+              limit: remainingLimit,
+              offset: adjustedOffset
+            });
 
-              if (getUserAgentType(req.headers["user-agent"]) !== UserAgentType.K8_OPERATOR) {
-                await server.services.telemetry.sendPostHogEvents({
-                  event: PostHogEventTypes.SecretPulled,
-                  distinctId: getTelemetryDistinctId(req),
-                  properties: {
-                    numberOfSecrets: secretCountFromEnv,
-                    workspaceId: projectId,
-                    environment,
-                    secretPath,
-                    channel: getUserAgentType(req.headers["user-agent"]),
-                    ...req.auditLogInfo
+            for await (const environment of environments) {
+              const secretCountFromEnv = secrets.filter((secret) => secret.environment === environment).length;
+
+              if (secretCountFromEnv) {
+                await server.services.auditLog.createAuditLog({
+                  projectId,
+                  ...req.auditLogInfo,
+                  event: {
+                    type: EventType.GET_SECRETS,
+                    metadata: {
+                      environment,
+                      secretPath,
+                      numberOfSecrets: secretCountFromEnv
+                    }
                   }
                 });
+
+                if (getUserAgentType(req.headers["user-agent"]) !== UserAgentType.K8_OPERATOR) {
+                  await server.services.telemetry.sendPostHogEvents({
+                    event: PostHogEventTypes.SecretPulled,
+                    distinctId: getTelemetryDistinctId(req),
+                    properties: {
+                      numberOfSecrets: secretCountFromEnv,
+                      workspaceId: projectId,
+                      environment,
+                      secretPath,
+                      channel: getUserAgentType(req.headers["user-agent"]),
+                      ...req.auditLogInfo
+                    }
+                  });
+                }
               }
             }
           }
         }
+      } catch (error) {
+        if (!(error instanceof ForbiddenError)) {
```

Comment:
So if it is a instance of forbidden then it won’t throw error? How come

Reply:
It will be an empty list

Reply:
Why not throw the error? It feels like something the user should be notifed of

Reply:
It shouldn't. Think of this way. User coming to the root folder. They only have access to nested folder 

Now they have 4 folders and according to pagination it gets 20 per page. So the remaining 16 would be secrets. But the secrets will throw an error as permission doesn't allow thus the dashboard won't load and show the folders.

What happens the folder is not shown. It means they can't access the nested folders.

Reply:
Oh i see. It sounds like behavior similar to what we had before where if you didn't have access to secrets, it just souldn't show for that path